### ISSUE TITLE
Fix BlockDeletionGenerator poor documentation

### ISSUE THREAD

**Issue by WilkAndy** [idaholab/moose#17241]
## Reason

`BlockDeletionGenerator` has poor documentation, which caused multiple MOOSE failures for me recently (really annoyed by this, grumble, grumble).  It needs to be fixed.

## Design

```
  params.addParam<std::vector<SubdomainName>>("block", "The list of blocks to be deleted");
  params.addDeprecatedParam<SubdomainID>(
      "block_id", "The block to be deleted.", "Use block instead");
```
Should read
```
  params.addParam<std::vector<SubdomainName>>("block", "The list of block names to be deleted");
  params.addDeprecatedParam<SubdomainID>(
      "block_id", "The block ID to be deleted.", "Create names for each of your blocks, and use block instead");
```

(Aside: i don't understand why block_id needed to be deprecated, but i presume it was for a good reason).

> Mesh generator which removes elements from the specified subdomains
> The BlockDeletionGenerator is used to remove elements from a mesh matching a user provided subdomain ID

must somehow be aligned with `block`.  Users will ask "what is this *subdomain* thing?" since the interface only allows them to specify `block`.

## Impact

Code that is usable without having to read the C++ code.

**Comment by loganharbour** [comment]
The improvement here was two fold:

1. Allows for deletion of multiple blocks within the same generator
2. Switch from SubdomainID -> SubdomainName allows for the entry of subdomain by both ID and name. That is, `block = 1 some_block` would work.

You wrote that you would like the documentation to be "create names for each of your blocks". This is not the case. It still works with ID.

```
block = 1
```

and

```
block_id = 1
```

are the same.

I do agree, though - the deprecation message could be a little more detailed.

**Comment by WilkAndy** [comment]
Thanks @loganharbour .   I guess the fact that i thought `block` had to be a Name (by looking at the C++) illustrates the need for good doco.

**Comment by loganharbour** [comment]
> Thanks @loganharbour . I guess the fact that i thought `block` had to be a Name (by looking at the C++) illustrates the need for good doco.

Yeah, this is how we treat `block = ` in most places. For example in `BlockRestrictable` (enables block restriction for postprocessors, kernels, etc). It should be improved as well I think: https://github.com/idaholab/moose/blob/next/framework/src/interfaces/BlockRestrictable.C#L30
